src/domain/FineractClient/FineractClient.ts

```diff
@@ -25,275 +25,233 @@
+            },
+        });
+    }
+
+    private getDefaultHeaders() {
+        return {
+            'fineract-platform-tenantId': this.fineractConfig.FINERACT_TENANT_ID,
+            Authorization: this.getAuthHeader(),
+        };
     }
 
-    private getAuthHeader():string{
-        if(this.fineractConfig.FINERACT_AUTH_MODE == "oauth"){
+    private getAuthHeader(): string {
+        if (this.fineractConfig.FINERACT_AUTH_MODE == 'oauth') {
             // return oauth request header
         }
-        return `Basic ${Buffer.from(`${this.fineractConfig.FINERACT_USERNAME}:${this.fineractConfig.FINERACT_PASSWORD}`).toString("base64")}`;
+        return `Basic ${Buffer.from(`${this.fineractConfig.FINERACT_USERNAME}:${this.fineractConfig.FINERACT_PASSWORD}`).toString('base64')}`;
     }
 
-    private async getClient(clientId: number): Promise<THttpResponse<TFineractGetClientResponse> | undefined>{
+    private async getClient(clientId: number): Promise<THttpResponse<TFineractGetClientResponse>> {
         const url = `${this.fineractConfig.FINERACT_BASE_URL}/${ROUTES.clients}/${clientId}`;
         this.logger.info(`Request to fineract ${url}`);
-        const res = await this.httpClient.send<TFineractGetClientResponse>(
-            url,
-            {
-                method: "GET",
-                headers: {
-                    "fineract-platform-tenantId": this.fineractConfig.FINERACT_TENTANT_ID,
-                    "Authorization": this.getAuthHeader()
-                }
-            }
-        );
-        return res;
+        return await this.httpClient.send<TFineractGetClientResponse>({
+            url: url,
+            method: 'GET',
+            headers: {
+                ...this.getDefaultHeaders(),
+            },
+        });
     }
 
-    private async sendTransfer(transactionPayload: TFineracttransferDeps ): Promise<THttpResponse<TFineractTransactionResponse> | undefined>{
-        const url = `${this.fineractConfig.FINERACT_BASE_URL}/${ROUTES.savingsAccount}/${transactionPayload.accountId}/transactions?command=deposit`;
-        this.logger.info(`Request to fineract ${url}`);
-        const res = await this.httpClient.send<TFineractTransactionResponse>(
-            url,
-            {
-                method:"POST",
+    async sendTransfer(
+        transactionPayload: TFineractTransferDeps,
+    ): Promise<THttpResponse<TFineractTransactionResponse>> {
+        this.logger.info('Request to fineract. Withdraw');
+        const url = `${this.fineractConfig.FINERACT_BASE_URL}/${ROUTES.savingsAccount}/${transactionPayload.accountId}/transactions?command=withdrawal`;
+        try {
+            return await this.httpClient.post<TFineractTransactionPayload, TFineractTransactionResponse>(
+                url,
+                transactionPayload.transaction,
+                {
+                    headers: {
+                        ...this.getDefaultHeaders(),
+                        'Content-Type': 'application/json',
```

Comment:
shouldn't `'Content-Type': 'application/json'` be a part of  getDefaultHeaders()?